src/js/components/agencyV2/visualizations/StatusOfFundsChart.jsx

```diff
@@ -0,0 +1,141 @@
+    }, []);
+
+    // extract amounts to get max value for x axis and store agency names in order
+    for (let i = 0; i < data.results.length; i++) {
+        resultNums = resultNums.concat(data.results[i].total_budgetary_resources, data.results[i].total_obligations);
+        resultNames = resultNames.concat(data.results[i].name);
+    }
+    const sortedNums = resultNums.sort((a, b) => b - a);
+    // format dollar amounts
+    for (let i = 0; i < sortedNums.length; i++) {
+        formattedAmounts.push(MoneyFormatter.formatMoneyWithUnitsShortLabel(sortedNums[i]));
+    }
+    // Wrap y axis labels - reference https://bl.ocks.org/mbostock/7555321
+    function wrapText(selection) {
+        selection.each(function w() {
+            const node = d3.select(this);
+            const rectWidth = 500;
+            let word;
+            const words = node.text().split(' ').reverse();
+            let line = [];
+            const x = node.attr('x');
+            const y = node.attr('y');
+            let tspan = node.text('').append('tspan').attr('x', x).attr('y', y);
+            let lineNumber = 0;
+            function addTspan(text) {
+                lineNumber += 1;
+                return (
+                    node
+                        .append('tspan')
+                        .attr('x', x)
+                        .attr('y', y)
+                        .attr('dy', `${lineNumber * 16}px`)
+                        .text(text)
+                );
+            }
+            while (words.length > 1) {
+                word = words.pop();
+                line.push(word);
+                tspan.text(line.join(' '));
+                const tspanLength = tspan.node().getComputedTextLength();
+                if (tspanLength > rectWidth && line.length !== 1) {
+                    line.pop();
+                    tspan.text(line.join(' '));
+                    line = [word];
+                    tspan = addTspan(word);
+                }
+            }
+            addTspan(words.pop());
+        });
+    }
+    // prevent eslint error for conditional function call to wrapText()
+    function wrapTextMobile() {
+        return '';
+    }
+    // setup x and y scales
+    const y = scaleBand()
+        .range([0, isMobile ? chartHeight + 400 : chartHeight])
+        .padding(0.1);
+    const x = scaleLinear()
+        .range([0, isMobile ? chartWidth + 330 : chartWidth]);
+
+    // append the svg object to the div
+    d3.select('#sof_chart').selectAll('*').remove();
+    const svg = d3.select('#sof_chart')
+        .append('svg')
+        .attr('class', 'svg')
+        .attr("viewBox", [0, 0, viewWidth + margins.left + margins.right, viewHeight + margins.top + margins.bottom])
+        .append('g')
+        .attr('transform', `translate(${isMobile ? margins.left - 40 : margins.left}, ${margins.top})`);
+    // scale to x and y data points
+    x.domain([0, sortedNums[0]]);
+    y.domain(resultNames);
+    const tickNum = isMobile ? 3 : 5;
```

Comment:
I'm not seeing it switch to 3 ticks in mobile, but that could just be the mock data!

Reply:
Updated to address all comments!